core/serialization/deserializer/montage-deserializer.js

```diff
@@ -76,9 +95,39 @@ var MontageDeserializer = exports.MontageDeserializer = Montage.specialize({
 
     preloadModules: {
         value: function () {
-            var serialization = JSON.parse(this._serializationString);
+            var serialization = JSON.parse(this._serializationString),
+                reviver = this._reviver,
+                moduleLoader = reviver.moduleLoader,
+                object,
+                locationId,
+                locationDesc,
+                module,
+                promises = [];
+
+            for (var label in serialization) {
+                if (serialization.hasOwnProperty(label)) {
```

Comment:
Since that loop only does something when label is a property on serialization directly, it would probably be faster to just loop on Object.keys() which returns an array of a given object's own enumerable properties.